pkg/pull/signature.go

```diff
@@ -48,16 +68,16 @@ func VerifySignature(license *kotsv1beta1.License) error {
 		return errors.New("unknown global key")
 	}
 
-	if err := verify([]byte(signature.PublicKey), keySignature.Signature, globalKeyPEM); err != nil {
+	if err := verify([]byte(innerSignature.PublicKey), keySignature.Signature, globalKeyPEM); err != nil {
 		return errors.Wrap(err, "failed to verify key signature")
 	}
 
-	licenseMessage, err := getMessageFromLicense(license)
+	licenseMessage, err := base64.StdEncoding.DecodeString(outerSignature.LicenseData)
 	if err != nil {
-		return errors.Wrap(err, "failed to convert license to message")
+		return errors.New("failed to decode license data from outer signature")
 	}
 
-	if err := verify(licenseMessage, signature.LicenseSignature, []byte(signature.PublicKey)); err != nil {
+	if err := verify(licenseMessage, innerSignature.LicenseSignature, []byte(innerSignature.PublicKey)); err != nil {
```

Comment:
this verify can be run against the base64 encoded data itself
but really what matters is doing it the same way everywhere, I suppose
```suggestion
	if err := verify(outerSignature.LicenseData, innerSignature.LicenseSignature, []byte(innerSignature.PublicKey)); err != nil {
```

Reply:
we sign the data before encoding it, we can update both if that's the preferred approach

Reply:
If it's easier to sign before encoding, we can do that
(I don't have a preference here)